src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -34,9 +35,16 @@
 (defrecord AndExpression [clauses])
 (defrecord OrExpression [clauses])
 (defrecord NotExpression [clause])
+(defrecord FnExpression [function column params args statement])
 
 (def json-agg-row (comp h/json-agg h/row-to-json))
-(def supported-fns #{"sum" "avg" "min" "max" "count"})
+(def pdb-fns->pg-fns
+  {"sum" "sum"
+   "avg" "avg"
+   "min" "min"
+   "max" "max"
+   "count" "count"
+   "to_string" "to_char"})
```

Comment:
I might just `def` the map-invert of this as well, with a `pg-fns->pdb-fns`, but not a big deal :)